sys/net/network_layer/ng_ipv6/netif/ng_ipv6_netif.c

```diff
@@ -38,49 +38,58 @@ static char addr_str[NG_IPV6_ADDR_MAX_STR_LEN];
 static ng_ipv6_addr_t *_add_addr_to_entry(ng_ipv6_netif_t *entry, const ng_ipv6_addr_t *addr,
                                           uint8_t prefix_len, uint8_t flags)
 {
+    ng_ipv6_netif_addr_t *free_entry;
+
     for (int i = 0; i < NG_IPV6_NETIF_ADDR_NUMOF; i++) {
         if (ng_ipv6_addr_equal(&(entry->addrs[i].addr), addr)) {
             return &(entry->addrs[i].addr);
         }
 
-        if (ng_ipv6_addr_is_unspecified(&(entry->addrs[i].addr))) {
-            memcpy(&(entry->addrs[i].addr), addr, sizeof(ng_ipv6_addr_t));
-            DEBUG("ipv6 netif: Added %s/%" PRIu8 " to interface %" PRIkernel_pid "\n",
-                  ng_ipv6_addr_to_str(addr_str, addr, sizeof(addr_str)),
-                  prefix_len, entry->pid);
+        if (ng_ipv6_addr_is_unspecified(&(entry->addrs[i].addr)) && !free_entry) {
+            free_entry = &(entry->addrs[i]);
+        }
+    }
 
-            entry->addrs[i].prefix_len = prefix_len;
-            entry->addrs[i].flags = flags;
+    if (!free_entry) {
```

Comment:
`free_entry` is not initialized with `NULL` in l41, leading to this line not being `NULL`, but something(TM) else, leading very likely to the segfault described in https://github.com/RIOT-OS/RIOT/pull/3044#issuecomment-104583401